pkg/services/ngalert/notifier/compat.go

```diff
@@ -83,10 +83,17 @@ func PostableToGettableGrafanaReceiver(r *apimodels.PostableGrafanaReceiver, pro
 	}
 	out.Settings = jsonBytes
 
+	if listOnly {
+		out.Settings = nil
+		out.DisableResolveMessage = false
+		out.SecureFields = nil
+		out.Provenance = apimodels.Provenance(models.ProvenanceNone)
```

Comment:
I think we should always return the correct provenance. This information does not bear any sensitive info. Same as DisableResolvedMessage

Reply:
I could see keeping the provenance in order to display it in the UI, but is there any need for DisableResolvedMessage? IMO we should be setting to default all the fields not necessary to list the receivers.

Reply:
> DisableResolvedMessage

I think you're right. That's settings anyway. It's just in a confusing place.

Reply:
Definitely a better approach, agreed.